### PR TITLE
 Add workspace-enabled example contracts (was PR:#44)

### PR DESCRIPTION
**Pull request by tenuki**
This PR creates a workspace where dependencies are defined and several smart-contracts as part of the same workspace, with inherited dependencies.

We tried different strategies on how to run the workspace contracts' build, and we chosen to try that build as a "whole" analogous to a contract in the test matrix.  The execution will be then cartesian product with the platforms.

Having the whole workspace-contracts in that way, we expect it to be all contracts in the workspace compiled at once.

We also included the suggestions given in the PR:44 before.

Was:
```
At the moment there are a few errors on cargo-contract preventing us to run it completely flawlessly, mainly : 
 * https://github.com/paritytech/cargo-contract/issues/1172
 * https://github.com/paritytech/cargo-contract/issues/1265
That is why we chosen to run the suggested build with `cargo build` instead. But left commented out what 
would be expected to be working in the future when above tickets get fixed.
```

Updated to:
Updated the CI/CD to what it should be: `cargo contract build ...` which will fail right now, but have no problems when the cargo-contract version is v 4.0 including: https://github.com/paritytech/cargo-contract/pull/1358 .

### CODE REVIEW

**Comment by tenuki** [comment]
Checked the actual code in the workspace examples and have no problems when ran with cargo-contract v 4.0.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,15 +72,50 @@ jobs:
...
+          dir_names: "true"
+          dir_names_exclude_current_dir: "true"
+          dir_names_max_depth: 2
+          files: |
+            workspace-contracts/**
+          files_ignore: |
+            **/.gitignore
+            **/*.md
+            multi-contract-caller/**
+            upgradeable-contracts/**
```
Comment (ascjones) [inline .github/workflows/ci.yml:91]: These are redundant?

**Review by ascjones — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tenuki) [reply .github/workflows/ci.yml:91]: Oh maybe, yes, haven't pay much attention to changed-files macro, but looks redundant, yes.

Comment (ascjones) [reply .github/workflows/ci.yml:91]: Please remove then we can merge

Comment (tenuki) [reply .github/workflows/ci.yml]: done! thanks!

**Pull request merged**